formats/protobuf/commonMain/src/kotlinx/serialization/protobuf/internal/Streams.kt

```diff
@@ -189,42 +200,33 @@ internal class ByteArrayOutput {
     }
 
     fun encodeVarint32(value: Int) {
-        ensureCapacity(5)
         // Fast-path: unrolled loop for single byte
+        ensureCapacity(5)
         if (value and 0x7F.inv() == 0) {
             array[position++] = value.toByte()
             return
         }
-        // Fast-path: unrolled loop for two bytes
-        var current = value
-        array[position++] = (current or 0x80).toByte()
-        current = current ushr 7
-        if (value and 0x7F.inv() == 0) {
-            array[position++] = value.toByte()
-            return
-        }
-        encodeVarint32SlowPath(current)
+        val length = varIntLength(value.toLong())
+        encodeVarint(value.toLong(), length)
+    }
+
+    fun encodeVarint64(value: Long) {
+        val length = varIntLength(value)
+        ensureCapacity(length + 1)
```

Comment:
Suggest do +1 inside `varIntLength` or `VAR_INT_LENGTHS` init block to match the name.

Reply:
Then it will slightly complicate the loop (have to do decrement), let's leave it as is